CMSIS/RTOS2/FreeRTOS/Source/cmsis_os2.c

```diff
@@ -861,8 +861,8 @@ uint32_t osThreadEnumerate (osThreadId_t *thread_array, uint32_t array_items) {
 #if (configSUPPORT_DYNAMIC_ALLOCATION == 1)
     task  = pvPortMalloc (count * sizeof(TaskStatus_t));
 #else
-    TaskStatus_t task_on_stack;
-    task = &task_on_stack;
+    TaskStatus_t task_on_stack[count];
```

Comment:
Are you sure about this? is there a bug on github or online about this? This does look like it might be correct given the dynamic allocation is count*sizeof(taskstatus_t) but i'm suprised this bug is in the code. We should upstream this if this is a real bug.

Reply:
So, it is not a bug from the upstream. After forking from the ARM-software/CMSIS-FreeRTOS, Steve made some modifications to hid dynamic memory alloc and free behind conditional compilation. Check out this [commit](https://github.com/ARM-software/CMSIS-FreeRTOS/commit/8371fa3b3f1f2a12e7b8e8c2442cd3988a6dcc9d#).

Reply:
@zhi Agreed.  My bad.